internal/storage/image.go

```diff
@@ -871,15 +873,25 @@ func pullImageImplementation(ctx context.Context, lookup *imageLookupService, st
 		return RegistryImageReference{}, err
 	}
 
-	manifestBytes, err := copy.Image(ctx, policyContext, destRef, srcRef, &copy.Options{
-		SourceCtx:        &srcSystemContext,
-		DestinationCtx:   options.DestinationCtx,
+	manifestBytes, err := ociartifact.NewStore(store.GraphRoot(), &srcSystemContext).PullManifest(ctx, srcRef, &ociartifact.PullOptions{CopyOptions: &libimage.CopyOptions{
 		OciDecryptConfig: options.OciDecryptConfig,
-		ProgressInterval: options.ProgressInterval,
 		Progress:         options.Progress,
-	})
+	}})
 	if err != nil {
-		return RegistryImageReference{}, err
+		if errors.Is(err, ociartifact.ErrIsAnImage) {
+			manifestBytes, err = copy.Image(ctx, policyContext, destRef, srcRef, &copy.Options{
+				SourceCtx:        &srcSystemContext,
+				DestinationCtx:   options.DestinationCtx,
+				OciDecryptConfig: options.OciDecryptConfig,
+				ProgressInterval: options.ProgressInterval,
+				Progress:         options.Progress,
+			})
+			if err != nil {
+				return RegistryImageReference{}, err
+			}
+		} else {
+			return RegistryImageReference{}, fmt.Errorf("unable to try puling possible OCI artifact: %w", err)
```

Comment:
nit: It can early-return.
```suggestion
		if !errors.Is(err, ociartifact.ErrIsAnImage) {
			return RegistryImageReference{}, fmt.Errorf("unable to try puling possible OCI artifact: %w", err)
		}
		manifestBytes, err = copy.Image(ctx, policyContext, destRef, srcRef, &copy.Options{
			SourceCtx:        &srcSystemContext,
			DestinationCtx:   options.DestinationCtx,
			OciDecryptConfig: options.OciDecryptConfig,
			ProgressInterval: options.ProgressInterval,
			Progress:         options.Progress,
		})
		if err != nil {
			return RegistryImageReference{}, err
		}
```